### PR TITLE
Updated _run functions to be compitable with Windows

### PR DESCRIPTION
**Pull request by Shawn-Guo-CN**
Updted the script building part in kilosort, kilosort2, ironclust, waveclus, spykingcircus and klusta. Thus they are compatible with Windows.

**Note:** I can only test with Klusta, as I cannot install the others on my Windows.

### CODE REVIEW

**Comment by alejoe91** [comment]
@Shawn-Guo-CN I think you have to remove the write calls or specify the path. I think the second option is better, so in all sorters that use shellscript we would need to check if windows and if not write the bin/bash stuff

**Comment by Shawn-Guo-CN** [comment]
> @Shawn-Guo-CN I think you have to remove the write calls or specify the path. I think the second option is better, so in all sorters that use shellscript we would need to check if windows and if not write the bin/bash stuff

@alejoe91 I just removed all write calls. Let me check it first and then specify paths?

**Comment by samuelgarcia** [comment]
THis looks good.

**Comment by alejoe91** [comment]
@Shawn-Guo-CN did you test a Matlab sorter as well? Otherwise I can check tomorrow before merging :)

File: spikesorters/kilosort/kilosort.py [hunk 1]
```diff
@@ -191,8 +190,8 @@ def _run(self, recording, output_folder):
                         cd {tmpdir}
                         matlab -nosplash -nodisplay -r kilosort_master
                     '''.format(tmpdir=output_folder)
-        shell_cmd = ShellScript(shell_cmd, script_path=str(output_folder / 'run_kilosort.sh'), keep_temp_files=True)
-        shell_cmd.write(str(output_folder / 'run_kilosort.sh'))
+        shell_cmd = ShellScript(shell_cmd, keep_temp_files=True)
+        shell_cmd.write()
```
Comment (alejoe91) [inline spikesorters/kilosort/kilosort.py]: remove

**Comment by Shawn-Guo-CN** [comment]
@alejoe91 

> @Shawn-Guo-CN did you test a Matlab sorter as well? Otherwise I can check tomorrow before merging :)

Sorry for late reply. I didn't test Matlab, as I don't have Matlab installed on my laptop.

File: spikesorters/waveclus/waveclus.py [hunk 2]
```diff
@@ -158,8 +157,8 @@ def _run(self, recording, output_folder):
                 cd {tmpdir}
                 matlab -nosplash -nodisplay -r run_waveclus
             '''.format(tmpdir=tmpdir)
-        shell_cmd = ShellScript(shell_cmd, script_path=str(tmpdir / 'run_waveclus.sh'), keep_temp_files=True)
-        shell_cmd.write(str(tmpdir / 'run_waveclus.sh'))
+        shell_cmd = ShellScript(shell_cmd, keep_temp_files=True)
+        shell_cmd.write()
```
[the remark this answers is not in the document]
Comment (Shawn-Guo-CN) [reply spikesorters/waveclus/waveclus.py]: Ummm, wired, these are removed in my fork repo but still here.

File: spikesorters/kilosort2/kilosort2.py [hunk 3]
```diff
@@ -190,8 +189,8 @@ def _run(self, recording, output_folder):
                         cd {tmpdir}
                         matlab -nosplash -nodisplay -r kilosort2_master
                     '''.format(tmpdir=output_folder)
-        shell_cmd = ShellScript(shell_cmd, script_path=str(output_folder / 'run_kilosort2.sh'), keep_temp_files=True)
-        shell_cmd.write(str(output_folder / 'run_kilosort2.sh'))
+        shell_cmd = ShellScript(shell_cmd, keep_temp_files=True)
+        shell_cmd.write()
```
[the remark this answers is not in the document]
Comment (Shawn-Guo-CN) [reply spikesorters/kilosort2/kilosort2.py]: You mean, remove `shell_cmd.write()`?

[the remark this answers is not in the document]
Comment (Shawn-Guo-CN) [reply spikesorters/kilosort2/kilosort2.py]: I just realised something tricky here. If users want to `cd` into a different volume on Win, `cd {tmpdir}` wouldn't work.

File: spikesorters/waveclus/waveclus.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Shawn-Guo-CN) [reply spikesorters/waveclus/waveclus.py]: It seems that there are some other operations in the above lines. So, can you change this part?

[the remark this answers is not in the document]
Comment (alejoe91) [reply spikesorters/waveclus/waveclus.py]: Which operations?

**Pull request merged**